### DISCUSSION TITLE
pnpm v9

### DISCUSSION THREAD

**Discussion by zkochan** [pnpm/pnpm#6628] [General]
This is a discussion about pnpm v9 breaking changes.

- [x] remove the ability to automatically convert lockfile v5 to v6
- [x] [link-workspace-packages](https://pnpm.io/npmrc#link-workspace-packages) should be `false` by default. So, only packages specified with the `workspace:` protocol will be linked from the workspace. This behaviour is less confusing then linking all dependencies. https://github.com/pnpm/pnpm/pull/7485
- [x] set `dedupe-injected-deps` to `true` by default (related PR https://github.com/pnpm/pnpm/pull/7416).
- [x] set `hoist-workspace-packages` to `true by default (related PR https://github.com/pnpm/pnpm/pull/7451)
- [x] https://github.com/pnpm/pnpm/issues/7321
- [ ] throw an error when `resolutions` or `pnpm` field are used in a non-root workspace project (now a warning is printed https://github.com/pnpm/pnpm/pull/6658)
- [ ] make the path to the CAFS shorter (remove the files subdirectory from the path). It should reduce the amount of times these errors happen: https://github.com/pnpm/pnpm/issues/2605
- ~move the default location of the virtual store from `node_modules/.pnpm` to `.pnpm` in order to reduce issues like this: https://github.com/pnpm/pnpm/issues/6830~ People voted against it: https://twitter.com/pnpmjs/status/1739962479097122843
- [x] the `requiresBuild` field should either be removed from the lockfile or be included always, regardless if the dependency is allowed to be built or not (related PR https://github.com/pnpm/pnpm/pull/7167)
- [ ] patchedDependencies in the lockfile should not contain paths. It is enough to just store their hashes
- [ ] implement https://github.com/pnpm/pnpm/issues/6693 again as a breaking change
- [x] improve the way side-effects cache is saved in index files.
- [x] don't remove node_modules when registry configuration changes

Related PR: https://github.com/pnpm/pnpm/pull/7459

**Comment by Mister-Hope** [comment]
It would be great if someone can pick https://github.com/pnpm/pnpm/issues/4982.

Examples in node docs are already using `node:` for core modules (e.g.: https://nodejs.org/docs/latest/api/synopsis.html), while this behavior is broken since the options is added.

I tried to raise the pr but I am not be able to find the related codes in the codebase, seems that when getting builtinModules via [`node:module`](https://nodejs.org/docs/latest/api/module.html#modulebuiltinmodules), we also need to manually check the ones with `node:` prefix
